### ISSUE TITLE
Enable offline use of embedding model

### ISSUE THREAD

**Issue by InAnYan** [InAnYan/jabref#87]
When JabRef with AI PR is run first, `djl` will download files in background in order to work with embedding models (I guess the PyTorch backend and embedding model).

JabRef already has some issues that it takes too much time to open, now it will open 5 minutes later more :cry: 

I think this is only one time job, but it should run in background

**Comment by koppor** [comment]
The user **has to be modified** and has to **agree** to that.

Assume a secret research company using JabRef. No outside communication allowed. Each outside communication immediately reported to the manager - if not on the "allow-list". Think of agreeing to a privacy policy putting something to the "allow-list". -- Maybe, that helps to understand why JabRef should not "just communicate" to the outside world.

Put differently: Companies have cooperate proxies which JabRef does not use out of the box. Thus, when starting  JabRef, there will be network exceptions in this case.

**Comment by InAnYan** [comment]
Fixed with some shit code. Or maybe it's not that bad

**Comment by koppor** [comment]
> Fixed with some shit code. Or maybe it's not that bad

You can link commits here. Otherwise, it is hard to follow.

Just pasting the commit id should work for linking

**Comment by koppor** [comment]
I think, I reopened, because I thought, I as suer was not informed, what's going on. I checked the help button

![image](https://github.com/user-attachments/assets/3b90bfbb-6e93-4b1c-80e3-96876c45749e)

It lead me to craft the comment https://github.com/JabRef/user-documentation/pull/491#issuecomment-2271439562.

I don't know about the proferences for local providers. Will there be a checkbox for enable chatting in general and then the local providers? - If yes, thenn a downlaod should only be made if external providers are enabled.

I think, with a split to local and remote AI #"providers", and the link to the help, it is OK as is.

Advanced: Add text "This will download embedding models to your machine".

**Comment by koppor** [comment]
Remember, I write somewhere about cooperate proxies. And machines not connected to the internet. How can an offline installation of JabRef work? How can users get these files?

**Comment by koppor** [comment]
Since the issue provides some interesting discussions, I keep it and just move it to the Week for offline LLMs.

**Comment by InAnYan** [comment]
DevCall: introduce online/offline mode

**Comment by koppor** [comment]
Folder which should maybe be copied from one machine to another: `C:\Users\vagrant\.djl.ai`.

**Comment by koppor** [comment]
Setting to try: Use [Qubes OS](https://www.qubes-os.org/) as operating system.

Scroll through https://www.qubes-os.org/screenshots/ to get an idea. Each color represents a separate VM.

**Comment by koppor** [comment]
At https://forum.image.sc/t/trouble-getting-gpu-to-work-with-instanseg-qupath/102042/26 I saw a nice download dialog. Maybe, we can get this, too?

![7b995ffd686eceb8c4ecaafefa7111d18c740023](https://github.com/user-attachments/assets/0e8a9bae-33a5-487c-b64b-480e71d651ae)

**Comment by InAnYan** [comment]
Closing, as moved here: https://github.com/JabRef/jabref/issues/12240
